### PR TITLE
Implemented login system

### LINKED ISSUE #23: Feature Request: Login system
**User Story**
For example: 
As a user
I want to be able to login to my account

**Priority**
[x]: High
[]: Medium
[]: Low

**Describe the solution you want to implement**
A login system that allows a user to access the application. A user will first need to register for an account before they are allowed to log in.

**Describe alternatives you've considered**
If a user doesn't have an account, they could log in using a default account.

**Issue Dependencies**
#22 
#19 

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by liappi**
Closes #23 

**Description**

Implemented login system.
- `user` table is accessed to compare password stored in DB and given password.

**Testing**

Manual testing on Postman. Logging in with the correct username and password is successful. Attempting to log in with a wrong username and password combination fails.

**Checklist**

- [X] I have tested my change
- [X] Code builds successfully and all tests pass
- [ ] Docs have been added/updated if needed

### CODE REVIEW

File: server/src/api/account.py [hunk 1]
```diff
@@ -47,9 +47,22 @@ def register(self):
...
+            username (string): Name used for login
+            password (string): Password used for login
         """
-        pass
+        JSON_object = json.loads(cherrypy.request.body.read().decode('utf-8'))
+        username = JSON_object.get("username")
+        password = JSON_object.get("password")
+
+        try:
+            isLoggedIn = self.database.get_password(username) == password
```
Comment (lken274) [inline server/src/api/account.py:63]: can you please store the username in a cherrypy session? that way the current user can be accessed from other endpoints

File: server/src/api/account.py [hunk 2]
```diff
@@ -47,9 +47,22 @@ def register(self):
...
+        except Exception as e:
+            print(e)
+        finally:
+            response = {
+                'success': isLoggedIn
```
Comment (lken274) [inline server/src/api/account.py:71]: what happens if get_password throws an exception? You're catching the exception but I don't think there's a default value for isLoggedIn when accessed from finally.

**Review by lken274 — CHANGES_REQUESTED** [review]

**Comment by lken274** [comment]
could you possibly change the title of the pr to specify "log in" instead of "logging"? I think there's some room for confusion with debug logging

**Review by tant836 — APPROVED** [review]
Approved pending changes suggested by Logan (lken274).

**Review by lken274 — APPROVED** [review]

**Pull request merged**